runtime/gc_modron_startup/mmparse.cpp

```diff
@@ -842,10 +927,16 @@ gcParseSovereignArguments(J9JavaVM *vm)
 	IDATA index = -1;
 	PORT_ACCESS_FROM_JAVAVM(vm);
 
-	if (!gcParseXlpOption(vm)) {
+	IDATA xlpOptionIndex = gcParseXlpOption(vm);
+	if (-1 == xlpOptionIndex) {
 		goto _error;
 	}
 
+	/* Setup to handle additional objectheap arguments */
+	if ((vm->largePageArgIndex != -1) && (vm->largePageArgIndex > xlpOptionIndex) && (!gcProcessXXLargePageObjectHeapArguments(vm))) {
```

Comment:
please put constant first `(-1 != vm->largePageArgIndex)`.

Reply:
With `xlpOptionIndex` can be -2 now the `(vm->largePageArgIndex > xlpOptionIndex)` condition looks confusing for case `vm->largePageArgIndex` is -1 `-1 > -2`. It is works just because it is guarded by `(vm->largePageArgIndex != -1)` as first condition. So it is correct code flow wise but has underwater mine for somebody might need change this code in the future

Reply:
As long as the return codes are correctly commented, then it shouldn't be an issue for anyone looking at in the future.

Reply:
Done. See https://github.com/eclipse/openj9/pull/7476/commits/f7d88ae9047fd371fb725b18ab7c63187e7765db

Reply:
The principle Dmitri is aiming for relates to "code is read more than it's written"  and "we have limited brain space, don't spend it on the trivial".

By making code obviously correct, we don't have worry or double check this when debugging though this path in the future.

Reply:
In this case I can purpose a better solution.
What we can do is simply revert the initial idea of returning the index and have the method return the boolean. The return value will return false if there has been an error in consuming the options. 

We can then add an argument to the method by reference, where it can be populated to either the index of the right-most consumed index if one exists, or -1 if no argument was consumed. 

That would very much simplify the comparison.